DIG-1177: Allow user to see all data for a given patient

OrdiNeu opened the pull request:
## Ticket(s)
[DIG-1177](https://candig.atlassian.net/browse/DIG-1177)
This also incidentally fixes two other PRs:
[DIG-922](https://candig.atlassian.net/browse/DIG-922)
[DIG-921](https://candig.atlassian.net/browse/DIG-921)

## Description
This implements a new tree structure to see all data for a selected patient. In addition, the patient table has been completely redone.

In addition, this fixes a bunch of linter errors that I introduced with #76 

## Screenshots (if appropriate)
### After PR
![image](https://github.com/CanDIG/candig-data-portal/assets/4656440/ce1c664a-2018-4b8b-a161-8026d93e1736)

## Types of Change(s)
- [x] 🪲 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] Prettier linter doesn't return errors
- [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc. 
- [ ] Locally tested
- [ ] Dev server tested
- [ ] Production tested when merging into stable/production branch

[DIG-1177]: https://candig.atlassian.net/browse/DIG-1177?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DIG-922]: https://candig.atlassian.net/browse/DIG-922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DIG-921]: https://candig.atlassian.net/browse/DIG-921?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

OrdiNeu committed:
Patient search + count per cohort now in

OrdiNeu committed:
Fix the patient view to count search query patients properly

OrdiNeu committed:
Further styling

OrdiNeu committed:
[WIP] Single patient view

OrdiNeu committed:
Fix the search

OrdiNeu committed:
Code style fixes

CourtneyGosselin reviewed (changes requested):
I would probably change the search/collapse all section to have less height it just seems tall compared to the patient info title. I would also change the black container broader to be the blue used around the other containers or the grey used around tables.

OrdiNeu committed:
Fix the styling a bit

OrdiNeu committed:
Shrink the search field

OrdiNeu committed:
Fix linter errors

OrdiNeu commented:
> I would probably change the search/collapse all section to have less height it just seems tall compared to the patient info title. I would also change the black container broader to be the blue used around the other containers or the grey used around tables.

I tried to fix the header by making the Patient Info label a little larger, and shrunk the search field as it was the largest component. I've switched the border colour as well.

Rebasing it caused a bunch of linter errors to show up for me, so I've fixed those as well.

OrdiNeu committed:
Fix a warning on the main page

OrdiNeu committed:
Order patient view so that first level components are first

OrdiNeu committed:
Fix the summary view with error scaling

CourtneyGosselin reviewed (approved):
LGTM
